liualgotrader/data/alpaca.py

```diff
@@ -32,12 +32,52 @@ def __init__(self):
             raise AssertionError(
                 "Failed to authenticate Alpaca RESTful client"
             )
+        # for requesting market snapshots by chunk of symbols
+        self.symbol_chunk_size = 1000
 
-    def get_symbols(self) -> List[Dict]:
+    def get_symbols(self) -> List[str]:
         if not self.alpaca_rest_client:
             raise AssertionError("Must call w/ authenticated Alpaca client")
 
-        return self.alpaca_rest_client.list_assets()
+        symbols = [
+            asset.symbol for asset in self.alpaca_rest_client.list_assets(
+                status='active', asset_class='us_equity'
+            ) if asset.tradable
+        ]
+        return symbols
+
+    def get_market_snapshot(self) -> List[Dict]:
+        # parse market snapshots per chunk of symbols
+        symbols = self.get_symbols()
+        raw_snapshots_dict = {}
+        for i in range(0, len(symbols), self.symbol_chunk_size):
+            chunked_symbols = symbols[i:i+self.symbol_chunk_size]
+            raw_snapshots_dict.update(self.alpaca_rest_client.get_snapshots(chunked_symbols))
+
+        # convert snapshot object for each symbol to a dict
+        market_snapshots = []
+        for ticker, ticker_snapshots in raw_snapshots_dict.items():
+            if ticker_snapshots is None:
+                continue
+            try:
+                processed_snapshot = {
```

Comment:
@riven314 if it's done in parallel, the worker thread that pulls the chunk can apply a filter function (passed externally), this would reduce the returned size as well.